feat: support for disabled tooltip callback

nareshpingale opened the pull request:
Addresse #1210 

Usage:
```
   <button
      data-tooltip-id="anchor-select"
      data-tooltip-content="this content is different"
      data-tooltip-dynamic="true"
      style={{
        maxWidth: '200px',
        overflow: 'hidden',
        textOverflow: 'ellipsis',
        whiteSpace: 'nowrap',
      }}
    >
      Anchor select with dynamic tooltip and a very long label to test overflow
    </button>
    <button data-tooltip-id="anchor-select" data-tooltip-dynamic="true">
      Anchor select with dynamic tooltip
    </button>
    <button data-tooltip-id="anchor-select">Anchor select without dynamic tooltip</button>
  </p>
  <Tooltip
    id="anchor-select"
    disableTooltip={(anchor) => {
      const { scrollWidth, offsetWidth } = anchor as HTMLElement
      if (anchor?.getAttribute('data-tooltip-dynamic') && scrollWidth <= offsetWidth) {
        return true
      }

      return false
    }}
  >
    Tooltip content
  </Tooltip>
```


https://github.com/user-attachments/assets/932b76f2-8733-465a-bf02-8970da36435c



<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
	- Introduced a new optional prop `disableTooltip` for the Tooltip component, allowing developers to conditionally disable tooltips for specific anchors.
	- Enhanced the TooltipController component to support the new `disableTooltip` functionality, improving control over tooltip display behavior.
  
- **Documentation**
	- Updated documentation to include the new `disableTooltip` function, providing guidance on controlling tooltip visibility based on anchor states.
  
- **Tests**
	- Added test cases to validate the behavior of the `disableTooltip` prop, ensuring tooltips render correctly based on the prop's return value.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

nareshpingale commented:
@gabrieljablonski do let me know if we need to update /examples for this. Also if everything looks fine at first glance will start adding changes to /docs.

gabrieljablonski commented on src/components/Tooltip/Tooltip.tsx:
```diff
@@ -459,6 +460,9 @@ const Tooltip = ({
     const elementRefs = new Set(anchorRefs)
 
     anchorsBySelect.forEach((anchor) => {
+      if(disableTooltip?.(anchor)){
+        return;
+      }
```
This separation between `anchorSelect` and `anchorId` is something we're planning to improve on for v6 (`anchorId` is already deprecated), but until then, we should probably do the check below as well.

Maybe like this:

```ts
    const anchorById = document.querySelector<HTMLElement>(`[id='${anchorId}']`)
    if (anchorById && !disableTooltip?.(anchorById)) {
      elementRefs.add({ current: anchorById })
    }
```

danielbarion replied:
Please add one space between `if` and `(` on line 463

gabrieljablonski reviewed (commented):
Looks good for a first pass. If you don't mind, please add it to the "options" table on the docs.

Also, if you could try adding a test for the new prop under this [tests file](https://github.com/ReactTooltip/react-tooltip/blob/master/src/test/tooltip-props.spec.js), we'd appreciate it. No need to waste too much time on it though if you can't figure it out.

When those are done we'll do a last check again and see if we catch anything. The feature is simple enough, but just to make sure.

nareshpingale commented:
- Added tests for disableTooltip returning true and false
- Addressed formatting issue raised by @danielbarion 
- Added disableTooltip check to anchorById
- Added disableTooltip in options in docs, although wasn't able to verify this as there is a peer dependency error by our docusaurus version for react 18

gabrieljablonski reviewed (approved):
LGTM. Merging soon.
